From 2dbd4fdf48efc235641779fe8349c9b1b542a3bd Mon Sep 17 00:00:00 2001
From: Zeng Heng <zengheng4@huawei.com>
Date: Mon, 28 Jul 2025 09:50:43 +0800
Subject: [PATCH 735/960] downstream: arm64/mpam: Ensure the err variable is
 initialized

Ensure that err variables are initialized to 0 to avoid returning
uninitialized values.

Fixes: bb66b4d115e5 ("arm_mpam: Add mpam_msmon_read() to read monitor value")
Signed-off-by: Zeng Heng <zengheng4@huawei.com>
Signed-off-by: Liu Shixin <liushixin2@huawei.com>
---
 drivers/platform/mpam/mpam_devices.c | 5 ++---
 include/linux/resctrl_mpam.h         | 4 ++--
 2 files changed, 4 insertions(+), 5 deletions(-)

diff --git a/drivers/platform/mpam/mpam_devices.c b/drivers/platform/mpam/mpam_devices.c
index 62a6d667f4e4..cb78184f202e 100644
--- a/drivers/platform/mpam/mpam_devices.c
+++ b/drivers/platform/mpam/mpam_devices.c
@@ -1055,7 +1055,7 @@ static void __ris_msmon_read(void *arg)
 
 static int _msmon_read(struct mpam_component *comp, struct mon_read *arg)
 {
-	int err, idx;
+	int err = 0, idx;
 	bool read_again;
 	u64 wait_jiffies;
 	struct mpam_msc *msc;
@@ -1626,7 +1626,6 @@ static int mpam_msc_setup_error_irq(struct mpam_msc *msc)
 
 	/* Allocate and initialise the percpu device pointer for PPI */
 	if (irq_is_percpu(irq))
-
 		return __setup_ppi(msc);
 
 	/* sanity check: shared interrupts can be routed anywhere? */
@@ -1696,7 +1695,7 @@ static int mpam_dt_parse_resource(struct mpam_msc *msc, struct device_node *np,
 
 static int mpam_dt_parse_resources(struct mpam_msc *msc, void *ignored)
 {
-	int err, num_ris = 0;
+	int err = 0, num_ris = 0;
 	const u32 *ris_idx_p;
 	struct device_node *iter, *np;
 
diff --git a/include/linux/resctrl_mpam.h b/include/linux/resctrl_mpam.h
index 449620cf04b3..9bd28c63068c 100644
--- a/include/linux/resctrl_mpam.h
+++ b/include/linux/resctrl_mpam.h
@@ -318,9 +318,9 @@ static inline u32 resctrl_get_config_index(u32 closid,
 	case CDP_NONE:
 		return closid;
 	case CDP_CODE:
-			return (closid * 2) + 1;
+		return (closid * 2) + 1;
 	case CDP_DATA:
-			return (closid * 2);
+		return (closid * 2);
 	}
 }
 
-- 
2.43.0

